plugins/workloads/pkg/info.go

```diff
@@ -123,250 +127,32 @@ func CreateResourceProviderFromAPI(ctx context.Context, kube kubernetes.Interfac
-		}
-		replicationController := formatControllers("ReplicationController", item.Name, item.Namespace, string(item.UID), item.GetObjectMeta().GetOwnerReferences(), containers, item.Annotations, item.Labels)
-		interfaces = append(interfaces, replicationController)
-	}
-
-	// ReplicaSet
-	replicationSetControllers, err := kube.AppsV1().ReplicaSets("").List(ctx, listOpts)
-	if err != nil {
-		logrus.Errorf("Error fetching ReplicaSets %v", err)
+		logrus.Errorf("Error while getting all TopControllers: %v", err)
 		return nil, err
 	}
 
-	for _, item := range replicationSetControllers.Items {
+	for _, workload := range workloads {
+		topController := workload.TopController
 		var containers []ContainerResult
-		for _, container := range item.Spec.Template.Spec.Containers {
-			containers = append(containers, formatContainer(container, corev1.ContainerStatus{}, item.Spec.Template.GetCreationTimestamp()))
-		}
-		replicationController := formatControllers("ReplicaSet", item.Name, item.Namespace, string(item.UID), item.GetObjectMeta().GetOwnerReferences(), containers, item.Annotations, item.Labels)
-		interfaces = append(interfaces, replicationController)
-	}
-
-	controllerMap := map[string]*ControllerResult{}
-	for idx := range interfaces {
-		controllerMap[interfaces[idx].UID] = &interfaces[idx]
-	}
-	topController := map[string]bool{}
-	children := map[string][]string{}
-	for _, pointer := range controllerMap {
-		lastParent := pointer
-		for lastParent.ParentUID != "" {
-			newParent := controllerMap[lastParent.ParentUID]
-			if newParent == nil {
-				break
+		podCount := float64(len(workload.Pods))
+		for _, pod := range workload.Pods {
+			// Convert the unstructured object to cluster.
+			var pd corev1.Pod
+			err = runtime.DefaultUnstructuredConverter.
+				FromUnstructured(pod.UnstructuredContent(), &pd)
+			if err != nil {
+				return nil, err
 			}
-			lastParent = newParent
-		}
-		topController[lastParent.UID] = true
-		if pointer.ParentUID != "" {
-			children[lastParent.UID] = append(children[lastParent.UID], pointer.UID)
-		}
-	}
-
-	type jobMetadata struct {
-		startTime time.Time
-		endTime   time.Time
-	}
-	cronChildren := map[string][]jobMetadata{}
-	// Jobs
-	jobs, err := kube.BatchV1().Jobs("").List(ctx, listOpts)
-	if err != nil {
-		logrus.Errorf("Error fetching Jobs %v", err)
-		return nil, err
-	}
-
-	for _, item := range jobs.Items {
-		var containers []ContainerResult
-		if item.Status.Active+item.Status.Failed > 0 {
-			continue
-		}
-		ownerUID := getOwnerUID(item.GetObjectMeta().GetOwnerReferences())
-		if ownerUID != "" {
-			ownerController := controllerMap[ownerUID]
-			if ownerController != nil {
-				if item.Status.StartTime != nil && item.Status.CompletionTime != nil {
-					cronChildren[ownerUID] = append(cronChildren[ownerUID], jobMetadata{
-						startTime: item.Status.StartTime.Time,
-						endTime:   item.Status.CompletionTime.Time,
-					})
-				}
-				continue
+			for _, ctn := range pd.Spec.Containers {
```

Comment:
I think we'll end up getting the same containers for each pod - we really just want to know what the container names are. 

You could use [`GetPodSpec`](https://github.com/FairwindsOps/controller-utils/blob/e9a28b6e43fcb89e2d304d5004566c8c5c383181/pkg/controller/pod.go#L20) to get the container names.